### PR TITLE
feat(bigquery): Add example for creating a connection

### PR DESCRIPTION
**Pull request by pratikmore0290**
## Description

Fixes #<ISSUE-NUMBER>

Note: If you are not associated with Google, open an issue for discussion before submitting a pull request.

## Checklist

**Readiness**

- [x] Yes, **merge** this PR after it is approved
- [ ] No, don't **merge** this PR after it is approved

**Style**

- [x] My sample follows the rules described for Terraform in the [Effective Samples style guide](https://googlecloudplatform.github.io/samples-style-guide/)
- [x] My sample follows the other requirements and best practices in the [Contributing
guide](https://github.com/terraform-google-modules/terraform-docs-samples/blob/main/CONTRIBUTING.md#other-requirements-and-best-practices)

**Testing**

- [x] I have performed tests described in the [Contributing guide](https://github.com/terraform-google-modules/terraform-docs-samples/blob/main/CONTRIBUTING.md):

   - [x] **[Tests](https://github.com/terraform-google-modules/terraform-docs-samples/blob/main/CONTRIBUTING.md#set-up-the-test-environment)** pass: `terraform apply`
   - [x] **[Lint](https://github.com/terraform-google-modules/terraform-docs-samples/blob/main/CONTRIBUTING.md#linting-and-formatting)** pass: `terraform fmt` check

**Intended location**

- [x] Yes, this sample will be (or already is) included on cloud.google.com
      Location(s): https://cloud.google.com/bigquery/docs/create-cloud-resource-connection#create-cloud-resource-connection

- [ ] No, this sample won't be included on cloud.google.com
      Reason:

**API enablement**

- [ ] If the sample needs an API enabled to pass testing, I have added the service to the [Test setup file](https://github.com/terraform-google-modules/terraform-docs-samples/blob/main/test/setup/main.tf)

**Review**

- [ ] If this sample adds a new directory, I have added codeowners to the [CODEOWNERS file](https://github.com/terraform-google-modules/terraform-docs-samples/blob/main/.github/CODEOWNERS)

### CODE REVIEW

**Review by iennae — COMMENTED** [review]
This feels insufficient even though it's the base resource to describe how to create a connection. Can you share what page this will end up on? I think based on the docs I just browsed, it feels like this could be a snippet but the file should have the full set of resources required. 

From https://cloud.google.com/bigquery/docs/connections-api-intro#connection_types it seems that there will be a specific external data source and so I'd expect this to have something like that.

**Comment by wj-chen** [comment]
> This feels insufficient even though it's the base resource to describe how to create a connection. Can you share what page this will end up on? I think based on the docs I just browsed, it feels like this could be a snippet but the file should have the full set of resources required.
> 
> From https://cloud.google.com/bigquery/docs/connections-api-intro#connection_types it seems that there will be a specific external data source and so I'd expect this to have something like that.

This corresponds to https://cloud.google.com/bigquery/docs/create-cloud-resource-connection#terraform.

@pratikmore0290 I'd rename the folder to `/bigquery_create_connection_cloud_resource` to mention the connection type

**Review by wj-chen — APPROVED** [review]

File: bigquery/bigquery_create_connection_cloud_resource/main.tf [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+# [START bigquery_create_connection_cloud_resource]
+## This creates a cloud resource connection.
+## Note: The cloud resource nested object has only one output field - serviceAccountId.
+resource "google_bigquery_connection" "connection" {
+  connection_id = "my_cloud_resource_connection"
+  project       = "myproject1-381000"
```
Comment (glasnt) [inline bigquery/bigquery_create_connection_cloud_resource/main.tf:22]: You cannot hardcode a project into the sample. If you must provide a project argument, you should use `data "google_project" "default" {}`

File: bigquery/bigquery_create_connection_cloud_resource/main.tf [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+# [START bigquery_create_connection_cloud_resource]
+## This creates a cloud resource connection.
+## Note: The cloud resource nested object has only one output field - serviceAccountId.
+resource "google_bigquery_connection" "connection" {
```
Comment (glasnt) [inline bigquery/bigquery_create_connection_cloud_resource/main.tf:20]: The default resource name should be "default"
```suggestion
resource "google_bigquery_connection" "default" {
```

**Comment by glasnt** [comment]
Based on comments, I expect this CI validation to fail: /gcbrun

**Pull request merged**